platform-sdk/swirlds-common/src/main/java/com/swirlds/common/merkle/crypto/internal/MerkleInternalDigestProvider.java

```diff
@@ -65,7 +65,7 @@ protected Hash handleItem(
         for (int index = 0; index < childHashes.size(); index++) {
             final Hash childHash = childHashes.get(index);
 
-            if (childHash == null || childHash.getValue() == null) {
+            if (childHash == null || childHash.getBytes() == null) {
```

Comment:
Can the hash bytes (or the former Hash value) actually be `null`? We could ensure their non-null status in the Hash class itself and avoid this check here.

Reply:
It can be for now, until we get self serializable. When deserializing, we first construct an empty hash, then read the value. when the hash is constructed, its value is null. once we stop serializing hashes we can ensure that.